packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,324 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { HubFamily } from "../types";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+/**
+ * This is used to determin what IHubCatalog definition JSON object
+ * can be created
+ */
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+/**
+ * This is used to determin what IHubCollection definition JSON object
+ * can be created. We use HubFamily here to define most of the collections
+ * to ensure consistency
+ */
+export type WellKnownCollection =
+  | Exclude<HubFamily, "app" | "map">
+  | "appAndMap"
+  | "solution";
+
+/**
+ * A list of optional arguments to pass into getWellKnownCatalog
+ * user is the owner of the entity
+ * collectionNames is a list of names of the collections, only those collections
+ * in the catalog will be returned
+ */
+interface IGetWellKnownCatalogOptions {
+  user?: IUser;
+  collectionNames?: WellKnownCollection[];
+}
+
+/**
+ * Check if i18nScope is defined, if so add a "." at the end
```

Comment:
please add the "Why"... why are we adding a `.` to the end if it's defined?

Reply:
nit - this is not actually "validating" the i18nScope - rather it's just adding a `.` - not exactly sure what to call this fn - dotifyString? Also - I suggest adding a guard so that if the last char of the i18nScope is a `.`, don't add another `.`